data_structures/linked_list/print_reverse.py

```diff
@@ -8,48 +8,63 @@ def __init__(self, data=None):
 
     def __repr__(self):
         """Returns a visual representation of the node and all its following nodes."""
-        string_rep = ""
+        string_rep = []
         temp = self
         while temp:
-            string_rep += f"<{temp.data}> ---> "
+            string_rep.append(f"{temp.data}")
             temp = temp.next
-        string_rep += "<END>"
-        return string_rep
+        return "->".join(string_rep)
 
 
-def make_linked_list(elements_list):
+def make_linked_list(elements_list: List):
     """Creates a Linked List from the elements of the given sequence
-    (list/tuple) and returns the head of the Linked List."""
-
-    # if elements_list is empty
+    (list/tuple) and returns the head of the Linked List.
+    >>> make_linked_list([])
+    Traceback (most recent call last):
+        ...
+    Exception: The Elements List is empty
+    >>> make_linked_list([7])
+    7
+    >>> make_linked_list(['abc'])
+    abc
+    >>> make_linked_list([7, 25])
+    7->25
+    """
     if not elements_list:
         raise Exception("The Elements List is empty")
 
-    # Set first element as Head
-    head = Node(elements_list[0])
-    current = head
-    # Loop through elements from position 1
-    for data in elements_list[1:]:
-        current.next = Node(data)
+    current = head = Node(elements_list[0])
+    for i in range(1, len(elements_list)):
```

Comment:
@shellhub Thanks for the review and merge, I was hoping to improve the algorithm by using the generator as a part of my Hacktoberfest contribution for #2510. What was the reason for removing it for the regular iteration over a list?

i.e. (what I had before)

```
for data in (elements_list[i] for i in range(1, list_length)):
```

Reply:
This way need to generate a list and then iterate. It's slow.

Reply:
``` python
def fun1() -> None:
    my_list = []
    elements_list = [14, 52, 14, 12, 43]
    for i in range(0, len(elements_list)):
        my_list.append(i)


def fun2() -> None:
    my_list = []
    elements_list = [14, 52, 14, 12, 43]
    for data in (elements_list[i] for i in range(1, len(elements_list))):
        my_list.append(data)


if __name__ == '__main__':
    import timeit

    print(timeit.timeit("fun1()", setup="from __main__ import fun1", number=100000))
    print(timeit.timeit("fun2()", setup="from __main__ import fun2", number=100000))
```
output:
0.07830935700000001
0.10762060600000001

Reply:
Thanks, @shellhub that was informative! I only ran `timeit` against the list slicing. I should have checked the regular iteration as well. I guess it's also better to keep it simple! 😃

Reply:
How we Iris dataset from an online source. Read the dataset into a data structure, preferably
in a 2D array??? Anyone help me please its my assignment